asset-transfer-basic/application-typescript/src/app.ts

```diff
@@ -108,64 +105,130 @@ async function main() {
             // This type of transaction would only be run once by an application the first time it was started after it
             // deployed the first time. Any updates to the chaincode deployed later would likely not need to run
             // an "init" type function.
-            console.log('\n--> Submit Transaction: InitLedger, function creates the initial set of assets on the ledger');
+            console.log(
+                '\n--> Submit Transaction: InitLedger, function creates the initial set of assets on the ledger',
+            );
+
+            // Submit a transaction, blocking until the transaction has been committed on the ledger
             await contract.submitTransaction('InitLedger');
             console.log('*** Result: committed');
 
-            // Let's try a query type operation (function).
-            // This will be sent to just one peer and the results will be shown.
-            console.log('\n--> Evaluate Transaction: GetAllAssets, function returns all the current assets on the ledger');
-            let result = await contract.evaluateTransaction('GetAllAssets');
-            console.log(`*** Result: ${prettyJSONString(result.toString())}`);
+            console.log(
+                '\n--> Evaluate Transaction: GetAllAssets, function returns all the current assets on the ledger',
+            );
 
-            // Now let's try to submit a transaction.
-            // This will be sent to both peers and if both peers endorse the transaction, the endorsed proposal will be sent
-            // to the orderer to be committed by each of the peer's to the channel ledger.
-            console.log('\n--> Submit Transaction: CreateAsset, creates new asset with ID, color, owner, size, and appraisedValue arguments');
-            await contract.submitTransaction('CreateAsset', 'asset13', 'yellow', '5', 'Tom', '1300');
+            let result = await contract.evaluateTransaction('GetAllAssets');
+            console.log(`*** Result: ${Buffer.from(result).toString()}`);
+
+            console.log(
+                '\n--> Submit Transaction: CreateAsset, creates new asset with ID, color, owner, size, and appraisedValue arguments',
+            );
+
+            await contract.submitTransaction(
+                'CreateAsset',
+                'asset13',
+                'yellow',
+                '5',
+                'Tom',
+                '1300',
+            );
             console.log('*** Result: committed');
 
-            console.log('\n--> Evaluate Transaction: ReadAsset, function returns an asset with a given assetID');
+            console.log(
+                '\n--> Evaluate Transaction: ReadAsset, function returns an asset with a given assetID',
+            );
             result = await contract.evaluateTransaction('ReadAsset', 'asset13');
-            console.log(`*** Result: ${prettyJSONString(result.toString())}`);
+            console.log(`*** Result: ${Buffer.from(result).toString()}`);
```

Comment:
For the result, you probably want to use something like
```
Buffer.from(result, 'hex').toString()
```
ie add the hex parameter otherwise the output won't be easily understood

Reply:
Is the JSON returned by the chaincode really hex encoded? I thought it would be UTF-8 encoded to byte data.

Reply:
@davidkel I don't think this operation is possible on unit8array. Also output looks good with default  UTF-8 encoding .